packages/catalog-readmodel-writer/src/consumerServiceV2.ts

```diff
@@ -13,19 +13,46 @@ export async function handleMessageV2(
     .with({ type: "EServiceCloned" }, (msg) => msg.data.clonedEservice)
     .otherwise((msg) => msg.data.eservice);
 
-  await eservices.updateOne(
-    {
-      "data.id": message.stream_id,
-      "metadata.version": { $lt: message.version },
-    },
-    {
-      $set: {
-        data: eservice ? fromEServiceV2(eservice) : undefined,
-        metadata: {
-          version: message.version,
-        },
-      },
-    },
-    { upsert: true }
-  );
+  await match(message)
+    .with({ type: "EServiceDeleted" }, async (message) => {
+      await eservices.deleteOne({
+        "data.id": message.stream_id,
+        "metadata.version": { $lt: message.version },
+      });
+    })
+    .with(
+      { type: "EServiceAdded" },
+      { type: "DraftEServiceUpdated" },
+      { type: "EServiceCloned" },
+      { type: "EServiceDescriptorAdded" },
+      { type: "EServiceDraftDescriptorUpdated" },
+      { type: "EServiceDescriptorActivated" },
+      { type: "EServiceDescriptorArchived" },
+      { type: "EServiceDescriptorPublished" },
+      { type: "EServiceDescriptorSuspended" },
+      { type: "EServiceDescriptorDeleted" },
+      { type: "EServiceDescriptorInterfaceAdded" },
+      { type: "EServiceDescriptorDocumentAdded" },
+      { type: "EServiceDescriptorInterfaceUpdated" },
+      { type: "EServiceDescriptorDocumentUpdated" },
+      { type: "EServiceDescriptorInterfaceDeleted" },
+      { type: "EServiceDescriptorDocumentDeleted" },
+      async (message) =>
+        await eservices.updateOne(
+          {
+            "data.id": message.stream_id,
+            "metadata.version": { $lt: message.version },
+          },
+          {
+            $set: {
+              data: eservice ? fromEServiceV2(eservice) : undefined,
```

Comment:
I was checking the `proto` definitions and saw that eservice is present in all events except for `clone`:
<img width="290" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/ef280673-c665-427b-b490-2b0a3c3b6af8">

So also clone requires and ad-hoc handler probably - or, we could discuss renaming `clonedEservice` in `eservice` in  `EServiceClonedV2`? So that it is consistent with all other events cc @MalpenZibo

Reply:
@ecamellini I think you refer to [this](https://github.com/pagopa/interop-be-monorepo/pull/245/files#diff-c5d2b1ba7f2c512b326e54c6ba9954d88aebf5285f366c1ddb32a4d3cb799e3bR13).
Is that enough to handle the case you are mentioning?

Reply:
Oh, it is :)
Still, I would consider the rename to avoid the need for that ad-hoc piece of code for a single event (maybe to discuss as a separate task/PR?)

Reply:
Ok, I have just created a task for that.